addons/mail/models/mail_thread.py

```diff
@@ -443,13 +443,14 @@ def _check_can_update_message_content(self, messages):
         """" Checks that the current user can update the content of the message.
         Current heuristic is
 
-          * limited to note;
+          * limited to note and comment
           * if no tracking;
           * only for user generated content;
         """
+        comment_id = self.env['ir.model.data']._xmlid_to_res_id('mail.mt_comment')
         note_id = self.env['ir.model.data']._xmlid_to_res_id('mail.mt_note')
-        if any(message.subtype_id.id != note_id for message in messages):
-            raise exceptions.UserError(_("Only logged notes can have their content updated on model '%s'", self._name))
+        if any(message.subtype_id.id not in (comment_id, note_id) for message in messages):
```

Comment:
I don't know much about message subtypes but are there messages of `message_type` "comment" that are not of `subtype_id` comment_id or note_id?

Reply:
"comment" is user input, "notification" type is for generated messages, "email" for incoming emails (and currently mass mailing generating messages), "user_notification" for user-specific messages.

As "comment" messages are user entered messages, most of them will effectively be either comments, either notes. There is no constraint, so in some cases it could be whatever you want, notably when using the composer or automated actions where you can give what you want to 'message_post' or its various higher level API

Bisous !

Reply:
I have simplified code so check is just on `message_type` being `comment`